tests/test_GitRepository.cc

```diff
@@ -71,7 +74,7 @@ void create_testfiles(const std::filesystem::path& name, size_t nr_files,
 
 } // anonymous namespace
 
-TEST_CASE("GitRepository Wrapper Test all", "[GitWrapper]")
+TEST_CASE("GitRepository Wrapper Test all", "[GitRepository]")
```

Comment:
:thinking: 

```suggestion
TEST_CASE("GitRepository Test all", "[GitRepository]")
```

Additionally I would suggest to break this apart, for me there are far too many unrelated `SECTION`s, which makes debugging more hard. (Breaking apart also of course not in the PR)

Reply:
Ack.